disco/disco.go

```diff
@@ -266,7 +275,121 @@ func MessageSummary(m Message) string {
 		return fmt.Sprintf("pong tx=%x", m.TxID[:6])
 	case *CallMeMaybe:
 		return "call-me-maybe"
+	case *BindUDPRelayEndpoint:
+		return "bind-udp-relay-endpoint"
+	case *BindUDPRelayEndpointChallenge:
+		return "bind-udp-relay-endpoint-challenge"
+	case *BindUDPRelayEndpointAnswer:
+		return "bind-udp-relay-endpoint-answer"
 	default:
 		return fmt.Sprintf("%#v", m)
 	}
 }
+
+// BindUDPRelayHandshakeState represents the state of the 3-way bind handshake
+// between UDP relay client and UDP relay server. Its potential values include
+// those for both participants, UDP relay client and UDP relay server. A UDP
+// relay server implementation can be found in net/udprelay. This is currently
+// considered experimental.
+type BindUDPRelayHandshakeState int
+
+const (
+	// BindUDPRelayHandshakeStateInit represents the initial state prior to any
+	// message being transmitted.
+	BindUDPRelayHandshakeStateInit BindUDPRelayHandshakeState = iota
+	// BindUDPRelayHandshakeStateBindSent is a potential UDP relay client state
```

Comment:
bit of a nit, but I'd probably reword this a bit along the lines of:

BindUDPRelayHandshakeStateBindSent is the first client state after transmitting a BindUDPRelayEndpoint message to a UDP relay server.

BindUDPRelayHandshakeStateChallengeSent is the first server state after receiving a BindUDPRelayEndpoint and replying with a BindUDPRelayEndpointChallenge message.

and so on, essentially removing the "potential" and describing at least one side of the state chart edges.

Reply:
Updated in https://github.com/tailscale/tailscale/pull/15480/commits/8dbd4e6a6459ce8caf78114630c89f3f19976b83.